src/lib/components/Widget.js

```diff
@@ -72,7 +72,10 @@ export const Widget = React.forwardRef((props, forwardedRef) => {
   const ethersProviderContext = useContext(EthersProviderContext);
 
   const cache = useCache();
-  const near = useNear();
+  let near = useNear();
+  if(near && configs?.[0]?.chainId) {
+    near = near[configs?.[0]?.chainId]
+  }
```

Comment:
> useNear is used in multiple places. This is not a clear way of switching chainId (or networkId). Better to pass it to the hook as an optional arg, e.g. useNear(chainId)

Is there a way to pass an argument into a singleton hook? I can change the value via state by exposing a setter but that way it would return the `near` with that `chainId` for all future calls of the `useNear` hook.

Reply:
We may need to use non-singleton for `useNear` that relies on the singleton hook for multi-network near structure.